scripts/hook_injector.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+'''
+hook_injector.py
+
+    This example harnesses the power of
+    Manticore as a debugger during program analysis,
+    providing introspection of hooks injected through
+    user-specified pattern inputs.
+
+'''
+import re
+import argparse
+
+from manticore import Manticore
+from elftools.elf.elffile import ELFFile
+from elftools.elf.sections import SymbolTableSection
+
+
+class Introspector(object):
+    """
+    Introspector object is used to correctly parse
+    an ELF binary with a symbol table
+    """
+
+    def __init__(self, target_f, pattern):
```

Comment:
A class with one or no methods apart from `__init__` can be just a function. Side note: there's also a great talk about not writing classes in Python ( https://www.youtube.com/watch?v=o9pEzgHorH0 ).

Reply:
The function should probably just return what is in `self.functions` currently.